### DISCUSSION TITLE
Open multiple files at once with the file picker?

### DISCUSSION THREAD

**Discussion by cem-al** [helix-editor/helix#5881] [Q&A]
Is it possible to open multiple files at once with the file picker? This feature would be really useful 👍🏼

**Comment by David-Else** [comment]
Yes, sort of, if you are using the latest version built from master you can select a file with `alt return` and it is opened in the background so you can continue picking other files.

**Reply by CptPotato** [reply]
To add to that, `C-v` and `C-s` will open the file (or buffer) in a new vertical or horizontal split.

**Reply by pascalkuthe** [reply]
Only semi related : It would be pretty cool if we could generate multiple selections from a picker. For example in global search or in the workspace picker you could select multiple entries with tab (like in FZF) and then when you press one of the various keys that open the file, the file would be opened with all selected items selected (once we have multibuffers we could even support multi selections across multiple files)

**Reply by CptPotato** [reply]
> Only semi related : It would be pretty cool if we could generate multiple selections from a picker. For example in global search or in the workspace picker you could select multiple entries with tab (like in FZF) and then when you press one of the various keys that open the file, the file would be opened with all selected items selected (once we have multibuffers we could even support multi selections across multiple files)

I thought about this as well (selecting multiple files and opening them) and think it's a cool idea. Though, it might be a bit confusing if more shortcuts are introduced to the picker; I think the existing ones are already not easily discoverable. `Tab`/`Shift-Tab` is already in use to move up and down.

**Reply by pascalkuthe** [reply]
> > Only semi related : It would be pretty cool if we could generate multiple selections from a picker. For example in global search or in the workspace picker you could select multiple entries with tab (like in FZF) and then when you press one of the various keys that open the file, the file would be opened with all selected items selected (once we have multibuffers we could even support multi selections across multiple files)
> 
> I thought about this as well (selecting multiple files and opening them) and think it's a cool idea. Though, it might be a bit confusing if more shortcuts are introduced to the picker; I think the existing ones are already not easily discoverable. `Tab`/`Shift-Tab` is already in use to move up and down.

Yeah just for opening multiple files in a batch that wouldn't be worth it but the multiple selection part would be really useful and might justfy additional bindings. I suggested `tab` to stag consistent with `fzf` and `skim`. It's unfortunate that is already used.

**Reply by sibouras** [reply]
> the file would be opened with all selected items selected 

this just got merged #12192 and having it directly in the picker will enable global search replace without having to use external tools, with `tab` to select multiple files also having `<c-a>` to toggle all files like in telescope.nvim will be nice to have

**Reply by haxfn** [reply]
> > the file would be opened with all selected items selected
> 
> this just got merged #12192 and 

@sibouras I think this is unrelated to multi-open from picker

> having it directly in the picker will enable global search replace ..

I am not sure how #12192 which is about cli opening at multi selection points useful for "tabbing" from the file picker and multi-opening.

**Reply by sibouras** [reply]
imagine pressing `<space>/` for global search in workspace, then search for something, you get multiple results from the same file and you also get results from a different file, with `tab` you can select exactly the results you want to select similar to what `rg --vimgrep` does

**Reply by haxfn** [reply]
ah, I was only thinking of utilising file-finder-picker, the global-search-picker is also viable

fair, now suddenly this feature request seems more complicated

**Reply by dridk** [reply]
This would be enough ! 
```
:open file/data/*.py  

```

**Reply by xaviduds** [reply]
> This would be enough !
> 
> ```
> :open file/data/*.py  
> ```

This in my project with *.rs opens a file called *.rs * does the same

**Reply by marchbox** [reply]
> > This would be enough !
> > ```
> > :open file/data/*.py  
> > ```
> 
> This in my project with *.rs opens a file called *.rs * does the same

Yeah same for me, * is never treated as wildcard.
